board/designa/snappermx6/snappermx6.c

```diff
@@ -39,60 +40,75 @@ static iomux_v3_cfg_t const uart5_pads[] = {
 	MX6_PAD_GPIO_9__GPIO1_IO09	  | MUX_PAD_CTRL(NO_PAD_CTRL),
 };
 
-#ifdef CONFIG_FEC_MXC
-static iomux_v3_cfg_t const enet_pads[] = {
-	MX6_PAD_ENET_MDC__ENET_MDC              | MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_MDIO__ENET_MDIO            | MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_REF_CLK__ENET_TX_CLK    	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_TX_EN__ENET_TX_EN    	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_TXD0__ENET_TX_DATA0	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_TXD1__ENET_TX_DATA1   	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_KEY_ROW2__ENET_TX_DATA2		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_KEY_ROW0__ENET_TX_DATA3		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_GPIO_18__ENET_RX_CLK		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_CRS_DV__ENET_RX_EN    	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_RXD0__ENET_RX_DATA0	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_RXD1__ENET_RX_DATA1	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_KEY_COL2__ENET_RX_DATA2		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_KEY_COL0__ENET_RX_DATA3		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_ENET_RX_ER__ENET_RX_ER    	| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_KEY_ROW1__ENET_COL		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-	MX6_PAD_KEY_COL3__ENET_CRS		| MUX_PAD_CTRL(ENET_PAD_CTRL),
-
-	/* PHY reset */
-	MX6_PAD_KEY_COL1__GPIO4_IO08		| MUX_PAD_CTRL(NO_PAD_CTRL),
-};
-
-#ifdef CONFIG_MXC_SPI
-static iomux_v3_cfg_t const ecspi1_pads[] = {
-	MX6_PAD_EIM_EB2__GPIO2_IO30 | MUX_PAD_CTRL(SPI_PAD_CTRL),
-	MX6_PAD_EIM_D18__ECSPI1_MOSI | MUX_PAD_CTRL(SPI_PAD_CTRL),
-	MX6_PAD_EIM_D17__ECSPI1_MISO | MUX_PAD_CTRL(SPI_PAD_CTRL),
-	MX6_PAD_EIM_D16__ECSPI1_SCLK | MUX_PAD_CTRL(SPI_PAD_CTRL),
-};
-
-int board_spi_cs_gpio(unsigned bus, unsigned cs)
+static int salmon_version_read(void)
 {
-	return (bus == 0 && cs == 0) ? (IMX_GPIO_NR(2, 30)) : -1;
+	int retval = 0;
+	// We need to do this using a low-level read as this is used prior to the device
+	// tree being initialised, so the dm_ functions will not work
+	// We're on GPIO bank 2, pins 9 & 10
+
+	uint32_t gpio_state = readl(GPIO2_BASE_ADDR);
```

Comment:
This should use the GPIO driver...also there is 

gpio_request_list_by_name()
dm_gpio_get_values_as_int()

Reply:
Does this work for things that need to run prior to device tree set up? The trick here is that we're using this pin to modify the device tree, so it needs to be done very early?

Reply:
Do you have an SPL? If so, it could perhaps be done there?

If not, could it be done in pre-relocation code, so that DM is setup. Then you would have the right fdt for post-relocation

Reply:
Yes, we're using the U-Boot SPL